Container.cs

```diff
@@ -46,30 +46,83 @@ private void Update()
-                prefab.m_createRuining = false;
-                prefab.m_requireSurfaceMaps = false;
-                prefab.m_clipTerrain = false;
-                prefab.m_snapBuildingNodes = false;
-                prefab.m_placementStyle = ItemClass.Placement.Procedural;
-                prefab.m_useFixedHeight = true;
-                prefab.m_lowerTerrain = true;
-                prefab.m_availableIn = ItemClass.Availability.GameAndAsset;
+                ((TrainTrackAI)elevatedPrefab.m_netAI).m_elevatedInfo = elevatedPrefab;
+                elevatedPrefab.m_followTerrain = false;
+                elevatedPrefab.m_flattenTerrain = false;
+                elevatedPrefab.m_createGravel = false;
+                elevatedPrefab.m_createPavement = false;
+                elevatedPrefab.m_createRuining = false;
+                elevatedPrefab.m_requireSurfaceMaps = false;
+                elevatedPrefab.m_clipTerrain = false;
+                elevatedPrefab.m_snapBuildingNodes = false;
+                elevatedPrefab.m_placementStyle = ItemClass.Placement.Procedural;
+                elevatedPrefab.m_useFixedHeight = true;
+                elevatedPrefab.m_lowerTerrain = true;
+                elevatedPrefab.m_availableIn = ItemClass.Availability.GameAndAsset;
             });
-        }
+
+            newName = "Station Track Tunnel";
+            var tunnelPrefab = ClonePrefab(originalPrefab, newName);
+            if (tunnelPrefab == null)
+            {
+                UnityEngine.Debug.LogError("Couldn't clone prefab!");
+                return;
+            }
+
+            TrainTrackAI stationAI = elevatedPrefab.GetComponent<TrainTrackAI>();
+            TrainTrackTunnelAI tunnelAI = elevatedPrefab.gameObject.AddComponent<TrainTrackTunnelAI>();
+            tunnelAI.m_outsideConnection = stationAI.m_outsideConnection;
+            tunnelAI.m_constructionCost = 0;
+            tunnelAI.m_maintenanceCost = 0;
+            GameObject.DestroyImmediate(stationAI);
+
+            later(() =>
+            {
+                tunnelPrefab.m_clipTerrain = false;
+
+                tunnelPrefab.m_createGravel = false;
+                tunnelPrefab.m_createPavement = false;
+                tunnelPrefab.m_createRuining = false;
+
+                tunnelPrefab.m_flattenTerrain = false;
+                tunnelPrefab.m_followTerrain = false;
+
+                tunnelPrefab.m_intersectClass = null;
+
+                tunnelPrefab.m_minHeight = -1;
+
+                tunnelPrefab.m_requireSurfaceMaps = false;
+                tunnelPrefab.m_snapBuildingNodes = false;
+
+                tunnelPrefab.m_placementStyle = ItemClass.Placement.Procedural;
+                tunnelPrefab.m_useFixedHeight = true;
+                tunnelPrefab.m_lowerTerrain = false;
+                tunnelPrefab.m_availableIn = ItemClass.Availability.GameAndAsset;
+            });
+
+
+
+            TrainTrackAI ai = originalPrefab.GetComponent<TrainTrackAI>();
```

Comment:
That's not a good idea. That allows placing existing train stations where it wouldn't be possible originally. In such cases station track may become elevated one or tunnel.